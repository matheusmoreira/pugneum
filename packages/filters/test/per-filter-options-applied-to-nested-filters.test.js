const lex = require('pugneum-lexer');
const parse = require('pugneum-parser');
const apply = require('../').apply;

const filename = require('path').basename(__filename);

const customFilters = {
  first: function(str, options) {
    return options.wrap? 'FIRST\n' + str + '\nEND FIRST' : str;
  },
  second: function(str, options) {
    return options.wrap? 'SECOND\n' + str + '\nEND SECOND' : str;
  }
};

test('per filter options are applied, even to nested filters', () => {
  const source = `
p
  :first:second
    Will be wrapped in second.
`;

  const ast = parse(lex(source, {filename}), {
    filename,
    src: source,
  });

  const options = {
      second: {wrap: true},
  };

  const output = apply(ast, customFilters, options);
  expect(output).toMatchSnapshot();
});
