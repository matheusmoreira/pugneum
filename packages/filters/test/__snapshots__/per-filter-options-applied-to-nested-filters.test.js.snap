// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`per filter options are applied, even to nested filters 1`] = `
Object {
  "filename": "per-filter-options-applied-to-nested-filters.test.js",
  "line": 0,
  "nodes": Array [
    Object {
      "attributeBlocks": Array [],
      "attrs": Array [],
      "block": Object {
        "filename": "per-filter-options-applied-to-nested-filters.test.js",
        "line": 2,
        "nodes": Array [
          Object {
            "attrs": Array [],
            "block": Object {
              "filename": "per-filter-options-applied-to-nested-filters.test.js",
              "line": 3,
              "nodes": Array [
                Object {
                  "attrs": Array [],
                  "block": Object {
                    "filename": "per-filter-options-applied-to-nested-filters.test.js",
                    "line": 3,
                    "nodes": Array [
                      Object {
                        "column": 5,
                        "filename": "per-filter-options-applied-to-nested-filters.test.js",
                        "line": 4,
                        "type": "Text",
                        "val": "Will be wrapped in second.",
                      },
                    ],
                    "type": "Block",
                  },
                  "column": 9,
                  "filename": "per-filter-options-applied-to-nested-filters.test.js",
                  "line": 3,
                  "name": "second",
                  "type": "Text",
                  "val": "SECOND
Will be wrapped in second.
END SECOND",
                },
              ],
              "type": "Block",
            },
            "column": 3,
            "filename": "per-filter-options-applied-to-nested-filters.test.js",
            "line": 3,
            "name": "first",
            "type": "Text",
            "val": "SECOND
Will be wrapped in second.
END SECOND",
          },
        ],
        "type": "Block",
      },
      "column": 1,
      "filename": "per-filter-options-applied-to-nested-filters.test.js",
      "isInline": false,
      "line": 2,
      "name": "p",
      "selfClosing": false,
      "type": "Tag",
    },
  ],
  "type": "Block",
}
`;
